src/backend/functions/work-packages-edit.ts

```diff
@@ -0,0 +1,369 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import middy from '@middy/core';
+import jsonBodyParser from '@middy/http-json-body-parser';
+import httpErrorHandler from '@middy/http-error-handler';
+import validator from '@middy/validator';
+import { descBulletConverter } from './projects';
+import { Handler } from 'aws-lambda';
+import { PrismaClient } from '@prisma/client';
+import { buildSuccessResponse, DescriptionBullet, workPackageEditInputSchemaBody } from 'utils';
+
+const prisma = new PrismaClient();
+
+export const editWorkPackage: Handler = async ({ body }, _context) => {
+  const {
+    wbsElementId,
+    userId,
+    name,
+    crId,
+    startDate,
+    duration,
+    wbsElementIds,
+    expectedActivities,
+    deliverables
+  } = body;
+
+  // get the original work package so we can compare things
+  const originalWorkPackage = await prisma.work_Package.findUnique({
+    where: {
+      wbsElementId
+    },
+    include: {
+      wbsElement: true,
+      dependencies: true,
+      expectedActivities: true,
+      deliverables: true
+    }
+  });
+
+  // if it doesn't exist we error
+  if (originalWorkPackage === null) {
+    throw new TypeError('Work Package with given wbsElementId does not exist!');
+  }
+
+  let changes = [];
+  // get the changes or undefined for each of the fields
+  const nameChangeJson = createChangeJsonNonList(
+    'name',
+    originalWorkPackage.wbsElement.name,
+    name,
+    crId,
+    userId,
+    wbsElementId
+  );
+  const startDateChangeJson = createChangeJsonNonList(
+    'start date',
+    originalWorkPackage.startDate,
+    startDate,
```

Comment:
The json input received will give you the date as a string, and date objects are objects, so the current comparison will always add a change, and comparing date object equality can cause issues since they're objects. You may want to consider checking a formatted string or another representation of these dates.

Reply:
The date Object doesn't appear to have a `.equals()` method, but you can convert a date to "milliseconds since Jan 1, 1970 which can then be subtracted. I don't know if this is the best way to do it though. String comparison should work provided the strings are the same format outputs from a date Object.

Reply:
do we care about just the date or the time too?